### PR TITLE
feat : added authentication functionality

### PR DESCRIPTION
**Pull request by omkarpatke**
# Added Authentication Login/Signup 

[**Preview:**](https://deploy-preview-9--faster-cycles-store.netlify.app/)

**Files to check:**
- Signup.jsx
- SignIn.jsx
- App.js
- userAuth-context.js
- RequiresAuth.js

### CODE REVIEW

File: src/pages/SignIn/SignIn.jsx [hunk 1]
```diff
@@ -1,29 +1,65 @@
...
+      const response = await axios.post(`/api/auth/login`, {
+        email , password
+      });
+      if(response.status === 200){
+        localStorage.setItem("token", response.data.encodedToken);
+        setIsLogIn(true);
+        navigate(from , {replace:true});
+      } 
+    } catch (err) {
+        setError("The email you entered is not Registered. Please SignUp!");
```
Comment (kaustubh-jsr) [inline src/pages/SignIn/SignIn.jsx]: You cam also add a different error message for incorrect password. At present, even for a registered email, it shows 'Email not registered' error for wrong password.

**Review by kaustubh-jsr — APPROVED** [review]
The UI looks good, and the authentication works smoothly. Could not find any issues with the code, although have added a suggestion, for proper error message in case of incorrect password.

File: src/pages/SignIn/SignIn.jsx (hunk 1 above)
Comment (omkarpatke) [reply src/pages/SignIn/SignIn.jsx]: thanks for the suggestion will work on it.

**Review by nikhiljugale007 — COMMENTED** [review]
Code is good. Good to merge.
UI suggestion you can validate form before submitting like if we keep email  empty or email is  not of type email  then it should show that it is empty or not a proper email.

**Pull request merged**